### PR TITLE
Wandile/feature/api model classes

### PR DESCRIPTION
**Pull request by devwasabi2**
# Description

Please include a summary of the changes and the related issue. List any dependencies that are required for this change.

## Type of change

- I was integrating the changes Tawanda and Tashil made in the api. I was able to debug the transactions endpoints. I'l debug the debitOrder endpoints later today. I was also able to alter the amount sql columns, due to possible issues that may arise because of the currency we are using.

### CODE REVIEW

File: sql/account_balance.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+-- changeset devwasabi2:create-credit-account-transactions-total-view
+CREATE VIEW credit_account_total AS 
+SELECT 
+    credit_account_id,
+    SUM(transaction_amount)::NUMERIC(26, 3) AS total_amount
+FROM 
+    transaction
+GROUP BY 
+    credit_account_id;
+-- rollback DROP VIEW IF EXISTS credit_account_total;
```
Comment (ryanbasiltrickett) [inline sql/account_balance.sql]: Change this file views.sql

File: sql/debit_order.sql [hunk 2]
```diff
@@ -10,4 +10,15 @@ CREATE TABLE debit_order (
...
+-- changeset ryanbasiltrickett:add-debit-order-disabled
+ALTER TABLE debit_order
+ADD debit_order_disabled BOOLEAN NOT NULL;
+-- rollback ALTER TABLE debit_order DROP COLUMN debit_order_disabled
+
+-- changeset devwasabi2:alter-debit-order-amount
+ALTER TABLE debit_order
+ALTER COLUMN debit_order_amount TYPE BIGINT USING debit_order_amount::BIGINT,
+ALTER COLUMN debit_order_amount SET NOT NULL;
+-- rollback ALTER TABLE debit_order DROP COLUMN debit_order_amount
```
Comment (ryanbasiltrickett) [inline sql/debit_order.sql]: Maybe make it easier and just change the main table and we will just recreate the db from scratch

File: sql/loan.sql [hunk 3]
```diff
@@ -12,3 +12,9 @@ CREATE TABLE loan (
     loan_type loan_type_enum NOT NULL
 );
 -- rollback DROP TABLE loan
+
+-- changeset devwasabi2:alter-loan-amount
+ALTER TABLE loan
+ALTER COLUMN loan_amount TYPE BIGINT USING loan_amount::BIGINT,
+ALTER COLUMN loan_amount SET NOT NULL;
+-- rollback ALTER TABLE loan DROP COLUMN loan_amount
```
Comment (ryanbasiltrickett) [inline sql/loan.sql]: Maybe make it easier and just change the main table and we will just recreate the db from scratch

File: sql/loan_interest.sql [hunk 4]
```diff
@@ -10,3 +10,8 @@ CREATE TABLE loan_interest (
 );
 -- rollback DROP TABLE loan_interest
 
+-- changeset devwasabi2:alter-loan-interest-amount
+ALTER TABLE loan_interest
+ALTER COLUMN loan_interest_amount TYPE BIGINT USING loan_interest_amount::BIGINT,
+ALTER COLUMN loan_interest_amount SET NOT NULL;
+-- rollback ALTER TABLE loan_interest DROP COLUMN loan_interest_amount
```
Comment (ryanbasiltrickett) [inline sql/loan_interest.sql]: Maybe make it easier and just change the main table and we will just recreate the db from scratch

File: sql/references.sql [hunk 5]
```diff
@@ -48,4 +48,14 @@ ADD CONSTRAINT fk_db_order_transactions_debit_order_id FOREIGN KEY (debit_order_
...
+
+-- changeset ryanbasiltrickett:add-fk-constraints-account-delegation-account
+ALTER TABLE account_delegation
+ADD CONSTRAINT fk_account_delegation_account_id FOREIGN KEY (account_id) REFERENCES account(account_id);
+-- rollback ALTER TABLE account_delegation DROP CONSTRAINT fk_account_delegation_account_id
+
+-- changeset ryanbasiltrickett:add-fk-constraints-account-delegation-delegated-account
+ALTER TABLE account_delegation
+ADD CONSTRAINT fk_account_delegation_delegated_account_id FOREIGN KEY (delegated_account_id) REFERENCES account(account_id);
+-- rollback ALTER TABLE account_delegation DROP CONSTRAINT fk_account_delegation_delegated_account_id
```
Comment (ryanbasiltrickett) [inline sql/references.sql:61]: Maybe make it easier and just change the main table and we will just recreate the db from scratch

File: sql/transaction.sql [hunk 6]
```diff
@@ -15,4 +15,10 @@ CREATE TABLE transaction (
...
 );
--- rollback DROP TABLE transaction
\ No newline at end of file
+-- rollback DROP TABLE transaction
+
+-- changeset devwasabi2:alter-transaction_amount
+ALTER TABLE transaction
+ALTER COLUMN transaction_amount TYPE BIGINT USING transaction_amount::BIGINT,
+ALTER COLUMN transaction_amount SET NOT NULL;
+-- rollback ALTER TABLE transaction DROP COLUMN transaction_amount
```
Comment (ryanbasiltrickett) [inline sql/transaction.sql]: Maybe make it easier and just change the main table and we will just recreate the db from scratch

**Review by ryanbasiltrickett — CHANGES_REQUESTED** [review]
Just apply the following changes

File: sql/account_balance.sql (hunk 1 above)
Comment (devwasabi2) [reply sql/account_balance.sql]: yea, I forgot to change from numeric to bigint here as well

**Review by ThashilBBD — DISMISSED** [review]

**Review by ryanbasiltrickett — APPROVED** [review]
LGTM

**Pull request merged**